code/clue.js

```diff
@@ -1,59 +1,171 @@
+  name: 'Dumbbell',
+  weight: 5
+}
+const poison = {
+  name: 'Poison',
+  weight: 0.2
+}
+const axe = {
+  name: 'Axe',
+  weight: 6
+}
+const bat = {
+  name: 'Bat',
+  weight: 1.5
+}
+const trophy = {
+  name: 'Trophy',
+  weight: 4
+}
+const pistol = {
+  name: 'Pistol',
+  weight: 2
+}
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
-
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
-
+//ARRAYS FOR ALL THE SUSPECTS, WEAPONS & ROOMS
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
 
-const rooms = []
+const rooms = [
+  'Dining Room', 
+  'Conservatory', 
+  'Kitchen', 
+  'Study', 
+  'Library', 
+  'Billiard Room', 
+  'Lounge', 
+  'Ballroom', 
+  'Hall', 
+  'Spa', 
+  'Living Room', 
+  'Observatory', 
+  'Theatre', 
+  'Guest House', 
+  'Patio'
+]
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
 
-// This function will be invoked when you click on the killer card.
+//OBJECT THAT KEEPS THE MYSTERY.
+const mystery = {
+killer: '',
+weapon: '',
+room: ''
+}
+
+//FUNCTIONS FOR THE PICK CARD KILLER, WEAPON & ROOM.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age}`;
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`;
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`;
+}
+//Invoking the function when you click on the killer card:
+document.getElementById('killerCard').addEventListener('click', pickKiller);
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`;
```

Comment:
Here we can remove the backticks because there it's just the variable value we are assigning
`document.getElementById('weaponName').innerHTML = mystery.weapon.name;`

Reply:
Aha okay, fixed.